### DISCUSSION TITLE
Order / sort, collection item

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#9345] [Archive]
Good day!

I was wondering if someone knows how to order or sort a singleton collection.
like when you go to collections page it orders the items, in the same way as setup in the datamodel.. how to get the same result when doing a request to that collection, as when a request is made it will not give the data in this order..
is there a way to do this. ? 

kind regards.

**Comment by rijkvanzanten** [comment]
> as when a request is made it will not give the data in this order..

What order exactly? A singleton only returns a single item, so there's no sorting that one item?

**Reply by cpt-n3mo** [reply]
good day !

yeh sorry its abit difficult to explain as i think sort is something else..
what i mean to say is..
when one creates a data model, of a particular collection, one can drag and drop the items around. so that when one goes to the collection page the items in the collection have the same layout as given in the data model.. 
when one requests the data of that collection/item on a endpoint this layout data is not send with it.. is there a way to obtain this layout data somewere..?
at the moment im adding a string with the order of the items, so that i can order the items on a custom endpoint and be send back to the client. 

as a example..

one singleton collection `home` and i add 4 many to any relations in this order  `footer, banner, content, header` 
i drag the data around within the data model so it now hes this layout `header, banner, content, footer`

when one goes to the collection `home` one will find that the data hes the layout set by the data model,
but when one requests this data via api, the data hes the default order of how it was added in the first place, ignoring the layout set.., 

so the the question is how does one get to this layout data with the api..?
the reason for this is that dynamic components can render in the right order on the frontend. 

i hope this make the question more clear. :)

kind regards.

**Reply by rijkvanzanten** [reply]
Ahhh yes that makes sense! Unfortunately, the order of properties in JSON isn't guaranteed in most languages / tools, so I would recommend against relying on the key order in the JSON object itself.. That being said, you could fetch the fields configuration itself from the `/fields/:collection` endpoint, and get the reliable sort order through their `meta.sort` values 👍🏻

**Reply by cpt-n3mo** [reply]
ah Perfect!
This is exactly what i was searching for !
now i can group my components as well. and easily get the `meta.sort` from a custom endpoint.

```
const { FieldsService } = services;
const fields = new FieldsService({
      schema: req.schema,
      accountability: req.accountability,
})
fields.readAll('home').then((results)=>{
      const components = results.filter((e)=> e.meta.group === 'components')
      console.log(components) 
 })

```
I did not know that key order is not guaranteed inside a JSON so this is very good to know, i will adapt my idea accordingly thank you for the heads up, as this probably would have given me a major headegg later on <3 

kind regards
